crates/core/app/src/app/mod.rs

```diff
@@ -113,16 +120,26 @@ impl App {
                         start_height: 0,
                     },
                 );
+
+                Distributions::init_chain(&mut state_tx, Some(&())).await;
+                ShieldedPool::init_chain(&mut state_tx, Some(&app_state.shielded_pool_content))
+                    .await;
+                Staking::init_chain(&mut state_tx, Some(&app_state.stake_content)).await;
+                IBCComponent::init_chain(&mut state_tx, Some(&())).await;
+                Dex::init_chain(&mut state_tx, Some(&())).await;
+                Governance::init_chain(&mut state_tx, Some(&())).await;
```

Comment:
If the component parameters were set by the component itself, we'd need each of these impls to have the associated type be a container that had the component parameters (left open for other genesis data in the future), right?

That would involve making a proto message for each component's genesis data, which would be some amount of boilerplate, and we could definitely leave it out of this PR, but wanted to raise it for discussion or as a next step.

Reply:
Oh, wait, it looks like there are already `genesis::Content` types in each component?